keps/sig-node/4817-resource-claim-device-status/README.md

```diff
@@ -0,0 +1,731 @@
+    DNS        DNS         `json:"dns,omitempty"`
+    DeviceInfo *DeviceInfo `json:"device-info,omitempty"`
+    Gateway    []string    `json:"gateway,omitempty"`
+}
+```
+
+### Pod.Status.PodIPs Enhancement
+
+As part of the [Multi-Network (KEP-3698)](https://github.com/kubernetes/enhancements/issues/3698), 
+the idea was to use the existing `Pod.Status.PodIPs` and save the data about the
+different network interfaces/devices attached to the `Pod`. As part of the
+review of the KEP, it has been indicated ([here](https://github.com/kubernetes/enhancements/pull/3700#discussion_r1501690793) 
+and [here](https://github.com/kubernetes/kubernetes/pull/123112#issuecomment-1925957930))
+that it would be an API breaking change if the `Pod.Status.PodIPs` contains
+more than 1 value per IP family.
+
+### New Pod.Status Field
+
+Still as part of the [KEP-3698 - Multi-Network](https://github.com/kubernetes/enhancements/issues/3698), 
+and in the continuation of the previous alternative, the idea was to add a new field
+`Networks` in the `Pod.Status` so each networking DRA driver could report the
+status for each network interface/device directly in the `Pod.Status`.
+
+Here is below the proposed API:
+```golang
+// PodStatus represents information about the status of a pod. Status may trail the actual
+// state of a system, especially if the node that hosts the pod cannot contact the control
+// plane.
+type PodStatus struct {
+    ...
+
+    // Networks is a list of PodNetworks that are attached to the Pod.
+    //
+    // +optional
+    Networks []NetworkStatus `json:"networks,omitempty"`
+}
+
+// NetworkStatus provides the status of specific PodNetwork in a Pod.
+type NetworkStatus struct {
+    // Name is name of PodNetwork
+    Name string `json:"name"`
+
+    // InterfaceName is the network interface name inside the Pod for this attachment.
+    // Examples: eth1 or net1
+    //
+    // +optional
+    InterfaceName string `json:"interfaceName"`
+
+    // ip is an IP address (IPv4 or IPv6) assigned to the pod
+    IP string `json:"ip,omitempty"`
+
+    // IsDefaultGW is a flag indicating that the interface with this IP
+    // inside the Pod holds the Default Gateway.
+    //
+    // +optional
+    IsDefaultGW bool `json:"isDefaultGW,omitempty"`
+}
+```
+
+### KEP-4680 extension
+
+During the WG Device Management meeting on 17th of September 2024 ([Slack
+summary](https://kubernetes.slack.com/archives/C0409NGC1TK/p1726679433650409)),
+the idea was to extend the [KEP-4680 about resource health status in the
+`Pod.Status`](https://github.com/kubernetes/enhancements/issues/4680) in order
+to expose device information and not just the health.
+
+### Custom Resources
+
+In the `ResourceClaim.Status.Devices`, instead of having opaque field (`Data`) and
+specific type fields, an object reference could be used for each device. The custom 
+object would be created and maintained by the driver to report the status of the 
+devices.
```

Comment:
I like this. It's very Gateway-API-ish (eg, [ParametersReference](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.ParametersReference))

Reply:
This would be my last choice, as if you want to build a controller (e.g. EndpointSlice controller) that gets the IPs for some pods, then you will need to find those pods, then get the ResouceClaims for those pods (current scenario with what is being proposed), then get the status object for those ResourceClaims.

The best would be the `New Pod.Status Field` alternative, but from what I understood in the [Multi-Network KEP](https://github.com/kubernetes/enhancements/pull/3700) and some other discussion (e.g. [here](https://kubernetes.slack.com/archives/C0409NGC1TK/p1726679433650409)), this is not possible.

Reply:
@thockin CRXes!